components/AddDonation.js

```diff
@@ -0,0 +1,136 @@
+import styled from "styled-components";
+import { useState } from "react";
+import Counter from "./Counter";
+
+export default function AddDonation({ createDonation }) {
+  const [toggleAddButton, setToggleAddButton] = useState(true);
+  const [count, setCount] = useState(0);
+  function handleClick() {
+    setToggleAddButton(!toggleAddButton);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    createDonation(data);
+    setCount(0);
+    event.target.reset();
+    setToggleAddButton(!toggleAddButton);
```

Comment:
very nice separation of functionality here. The creation of the Donation is defined on a higher level, everything regarding the form is handled here locally. Great!

Reply:
😃